test/integration/instance.test.js

```diff
@@ -513,6 +513,49 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
       });
     });
 
+    it('should update internal options of the instance', function() {
+      var Book = this.sequelize.define('Book', { title: DataTypes.STRING })
+        , Page = this.sequelize.define('Page', { content: DataTypes.TEXT });
+
+      Book.hasMany(Page);
+      Page.belongsTo(Book);
+
+      return Book.sync({force: true}).then(function() {
+        return Page.sync({force: true}).then(function() {
+          return Book.create({ title: 'A very old book' }).then(function(book) {
+            return Page.create().then(function(page) {
+              return book.setPages([page]).then(function() {
+                return Book.findOne({
+                  where: { id: book.id }
+                }).then(function(leBook) {
+                  var oldOptions = leBook.$options;
+                  return leBook.reload({
+                    include: [Page]
+                  }).then(function(leBook) {
+                    expect(oldOptions).not.to.equal(leBook.$options);
+                    expect(leBook.$options.include.length).to.equal(1);
+                    expect(leBook.Pages.length).to.equal(1);
+                    expect(leBook.get({plain: true}).Pages.length).to.equal(1);
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+
+    it('should return an error when reload fails', function() {
+      return this.User.create({ username: 'John Doe' }).then(function(user) {
+        return expect(user.reload({
+          where: { id: 2 }
+        })).to.be.rejectedWith(
+          Sequelize.InstanceError,
+          'Instance could not be reloaded because it does not exist anymore (find call returned null)'
+        );
+      });
+    });
```

Comment:
> ```reload``` should really not accept a where statement (but you don't have to fix this in this PR)

Why? Sometimes I need to reload instance with a new restricting params and check it for passing or not.
But if you think this feature unnecessary, I can remove cloning ```where``` or even prohibit accepting (throw an exception).